src/components/files/hooks/useGetFileDownloadUrl.ts

```diff
@@ -0,0 +1,45 @@
+import { useApolloClient } from '@apollo/client';
+import { useSnackbar } from 'notistack';
+import { useCallback } from 'react';
+import {
+  FileDownloadUrl_Directory_Fragment,
+  FileNodeDownloadUrlDocument,
+  FileNodeDownloadUrlQuery,
+} from '../FileActions';
+
+export const useGetFileDownloadUrl = () => {
+  const { enqueueSnackbar } = useSnackbar();
+
+  const showSnackbarError = useCallback(() => {
+    enqueueSnackbar('Could not download file', {
+      variant: 'error',
+    });
+  }, [enqueueSnackbar]);
+
+  const client = useApolloClient();
+
+  const getFileDownloadUrl = useCallback(
+    async (id: string): Promise<string> => {
+      try {
+        const { data } = await client.query<FileNodeDownloadUrlQuery>({
+          query: FileNodeDownloadUrlDocument,
+          variables: { id },
+          fetchPolicy: 'network-only',
+        });
+        const isDirectory = (
+          node: FileNodeDownloadUrlQuery['fileNode'] | undefined
+        ): node is FileDownloadUrl_Directory_Fragment => {
+          return node?.type === 'Directory';
+        };
+        return !isDirectory(data?.fileNode)
+          ? data?.fileNode.downloadUrl ?? ''
+          : '';
+      } catch {
+        showSnackbarError();
+        return '';
```

Comment:
I think we should probably use null/undefined for error cases. I typically don't think that empty string should be used to represent the absence of a value. There are some exceptions when displaying strings, but since this is a url it won't be displayed.